src/stepFunctions/commands/publishStateMachine.ts

```diff
@@ -20,20 +21,39 @@ import {
     PublishStateMachineWizardUpdateResponse,
 } from '../wizards/publishStateMachineWizard'
 
+import { VALID_SFN_PUBLISH_FORMATS, YAML_FORMATS } from '../constants/aslFormats'
+
 const DEFAULT_REGION: string = 'us-east-1'
 
 export async function publishStateMachine(awsContext: AwsContext, outputChannel: vscode.OutputChannel) {
     const logger: Logger = getLogger()
 
     const textDocument = vscode.window.activeTextEditor?.document
+
     if (!textDocument) {
         logger.error('Could not get active text editor for state machine definition')
         throw new Error('Could not get active text editor for state machine definition')
     }
 
-    if (textDocument.languageId === 'asl-yaml') {
-        logger.error('Cannot publish state machine from Amazon States Language YAML file')
-        throw new Error('Cannot publish state machine from Amazon States Language YAML file')
+    let text = textDocument.getText()
+
+    if (!VALID_SFN_PUBLISH_FORMATS.includes(textDocument.languageId)) {
+        const errorMessage = `Cannot publish state machine from "${textDocument.languageId}" file`
+        logger.error(errorMessage)
+        throw new Error(errorMessage)
+    }
+
+    if (YAML_FORMATS.includes(textDocument.languageId)) {
+        try {
+            text = JSON.stringify(safeLoad(text), undefined, '  ')
+        } catch (error) {
+            const message = 'Cannot publish invalid YAML file'
+            const localizedMsg = localize('AWS.message.error.stepFunctions.publishStateMachine.invalidYAML', message)
+
+            logger.error(error)
+            showErrorWithLogs(localizedMsg)
```

Comment:
No need for a separate const for the message, inline that with the localize call.